src/LoopDriver.php

```diff
@@ -7,20 +7,29 @@
     /**
      * Start the event loop.
      *
+     * The loop MUST continue to run until it is either stopped explicitly, no referenced watchers exist anymore, or an
+     * exception is thrown that cannot be handled. Exceptions that cannot be handled are exceptions thrown from an
+     * error handler or exceptions that would be passed to an error handler but non exists to handle them.
+     *
      * @return void
      */
     public function run();
 
     /**
      * Stop the event loop.
      *
+     * When an event loop is stopped, it continues with its current tick and exits the loop afterwards. Multiple calls
+     * to stop MUST be ignored and MUST NOT raise an exception.
+     *
      * @return void
      */
     public function stop();
 
     /**
      * Defer the execution of a callback.
      *
+     * The deferred callable MUST be executed in the next tick of the event loop.
```

Comment:
If the loop is executing other defer callbacks when another one is added, it *will* be executed in the same tick of the loop (with our current implementation). I'm not sure if it matters if it's the current tick or the next tick.